src/jvm/test/org/jetbrains/kotlinx/lincheck_test/util/TestUtils.kt

```diff
@@ -55,53 +54,88 @@ private fun compareAndOverwrite(expectedOutputFilePrefix: String, actualOutput:
     check(!expectedOutputFilePrefix.contains(".txt")) {
         "Filename $expectedOutputFilePrefix should not contain a file extension (.txt)"
     }
-    // Always overwrite jdk8 non-trace
-    if (testJdkVersion == TestJdkVersion.JDK_8 && !isInTraceDebuggerMode && OVERWRITE_REPRESENTATION_TESTS_OUTPUT) {
-        getExpectedLogFileFromSources(getFileNameFor(expectedOutputFilePrefix, TestJdkVersion.JDK_8, false)).writeText(actualOutput)
```

Comment:
When running the test on the default jdk, it would be nice if it created a file if it was not found. Might be cleaner to do that in `getExpectedLogFile`

Reply:
Oh, sorry, I did not realize it was for this purpose. 
I restored this behavior.

Reply:
Maybe also throw when there exists no file AND we are not in JDK17. To encourage first running and creating the default.

Reply:
Let's allow creating the file automatically if the `OVERWRITE_REPRESENTATION_TESTS_OUTPUT` flag is passed.
It is easier this way to create the file on the first run.

A developer can then look at generated file or diff to verify it.

Reply:
Oh, sorry, I see your point now. 
If there is no "default" file, it will always create the jdk-specific file. Ok, I'll fix that.